src/usr/share/gpgmailer/gpgmailer-daemon.py

```diff
@@ -19,119 +19,255 @@
+
+    if not(signature_test(config['sender']['fingerprint'], config['sender']['password'], config['gpg_dir'])):
+        logger.warn('Sender key failed signature test.')
+        config['sender']['can_sign'] = False
+
+    else:
+        logger.debug('Sender key passed signature test.')
+        config['sender']['can_sign'] = True
+
+    for recipient in config['recipients']:
+        key_is_usable(recipient['fingerprint'], gpg_keyring)
+
+    global gpg_key_verifier
+    gpg_key_verifier = gpgkeyverifier.GpgKeyVerifier(gpg_keyring, config)
+
+    expiration_warning_message = gpg_key_verifier.get_expiration_warning_message(time.time())
 
-# Quit when SIGTERM is received
+    if expiration_warning_message:
+        message = 'Gpgmailer has just restarted. \n\n%s' % expiration_warning_message
+        mail_message = gpgmailmessage.GpgMailMessage()
+        mail_message.set_subject(config['default_subject'])
+        mail_message.set_body(message)
+        mail_message.queue_for_sending()
+        
+    logger.debug('Finished initial key check.')
+
+
+# Checks the sending key and configuration to determine if sending unsigned email
+#   is allowed. Crashes if the sending key cannot sign and sending unsigned email
+#   is disabled.
+#
+# config_dict: the dict to read key configuration from
+def verify_signing_config(config_dict):
+    if(not(config_dict['allow_expired_signing_key']) and not(config_dict['sender']['can_sign'])):
+        logger.critical('The sender key with fingerprint %s can not sign and \
+            unsigned email is not allowed. Exiting.' % config_dict['sender']['fingerprint'])
+        sys.exit(1)
+
+    elif not(config_dict['sender']['can_sign']):
+        logger.warn('The sending key is unable to sign. It may be expired or the password may be incorrect. Gpgmailer will send unsigned messages.')
+
+    else:
+        logger.debug('Outgoing emails will be signed.')
+
+
+# Quit when SIGTERM is received.
 def sig_term_handler(signal, stack_frame):
     logger.info("Quitting.")
     sys.exit(0)
 
-# TODO: Work out a permissions setup for gpgmailer so that it doesn't run as root.
+
+config, log_file_handle = build_config_dict()
+parse_key_config(config)
+
+gpg_keyring = gpgkeyring.GpgKeyRing(config['gpg_dir'])
+check_all_keys(config, gpg_keyring)
+
+verify_signing_config(config)
+
+# TODO: Eventually, check directory existence and permissions.
+# TODO: Eventually, move default outbox directory to /var/spool/gpgmailer
+
+logger.info('Verification complete.')
+
+
+# TODO: Eventually, either warn or crash when the config file is readable by everyone.
+# TODO: Eventually, work out a permissions setup for gpgmailer so that it doesn't run as root.
 daemon_context = daemon.DaemonContext(
     working_directory = '/',
-    pidfile = pidlockfile.PIDLockFile(PID_FILE),
+    pidfile = pidlockfile.PIDLockFile(pid_file),
     umask = 0
     )
 
-# TODO: Make a real cleanup method for this.
+# TODO: Eventually make a real cleanup method for this.
```

Comment:
I'm not sure. Are there things we need to clean up? Make sure there are no partially written or sent files or something?